E0195 needs to be updated to new format

From: src/test/compile-fail/E0195.rs

Error E0195 needs a span_label, updating it from:

```
error[E0195]: lifetime parameters or bounds on method `bar` do not match the trait declaration
  --> src/test/compile-fail/E0195.rs:18:5
   |
18 |     fn bar<'a,'b>(x: &'a str, y: &'b str) { //~ ERROR E0195
   |     ^
```

To:

```
error[E0195]: lifetime parameters or bounds on method `bar` do not match the trait declaration
  --> src/test/compile-fail/E0195.rs:18:5
   |
18 |     fn bar<'a,'b>(x: &'a str, y: &'b str) { //~ ERROR E0195
   |     ^ lifetimes do not match trait
```

Bonus: Underline the lifetimes and show the original trait requirement:

```
error[E0195]: lifetime parameters or bounds on method `bar` do not match the trait declaration
  --> src/test/compile-fail/E0195.rs:18:5
   |
13 |     fn bar<'a,'b:'a>(x: &'a str, y: &'b str);
   |           ---------- original trait requirement
...
18 |     fn bar<'a,'b>(x: &'a str, y: &'b str) { //~ ERROR E0195
   |           ^^^^^^^ lifetimes do not match trait
```

--- a/src/librustc_typeck/check/compare_method.rs
+++ b/src/librustc_typeck/check/compare_method.rs
@@ -411,10 +411,11 @@ pub fn compare_impl_method<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
         // are zero. Since I don't quite know how to phrase things at
         // the moment, give a kind of vague error message.
         if trait_params.len() != impl_params.len() {
-            span_err!(ccx.tcx.sess, span, E0195,
+            struct_span_err!(ccx.tcx.sess, span, E0195,
                 "lifetime parameters or bounds on method `{}` do \
-                         not match the trait declaration",
-                         impl_m.name);
+                 not match the trait declaration",impl_m.name)
+                .span_label(span, &format!("lifetimes do not match trait"))
+                .emit();
             return false;
         }
 

--- a/src/test/compile-fail/E0195.rs
+++ b/src/test/compile-fail/E0195.rs
@@ -16,6 +16,7 @@ struct Foo;
 
 impl Trait for Foo {
     fn bar<'a,'b>(x: &'a str, y: &'b str) { //~ ERROR E0195
+                                            //~^ lifetimes do not match trait
     }
 }
 

--- a/src/test/compile-fail/issue-16048.rs
+++ b/src/test/compile-fail/issue-16048.rs
@@ -29,6 +29,7 @@ impl<'a> Test<'a> for Foo<'a> {
 impl<'a> NoLifetime for Foo<'a> {
     fn get<'p, T : Test<'a>>(&self) -> T {
 //~^ ERROR E0195
+//~| lifetimes do not match trait
         return *self as T;
     }
 }

